src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -296,18 +296,39 @@ private void OnSourceProviderSourcesChanged(object sender, EventArgs e)
                         workspace, documentId, cancellationToken);
 
                     return TryInstallAndAddUndoAction(
-                        source, packageName, versionOpt, includePrerelease, dte, dteProject, undoManager);
+                        source, packageName, versionOpt, includePrerelease, projectId, dte, dteProject, undoManager);
                 }
             }
 
             return false;
         }
 
+        private bool IsPackageInstalled(ProjectId projectId, string packageName)
+        {
+            var cancellationToken = CancellationToken.None;
```

Comment:
we coudl consdier actually making this codepath cancellable with a TWD.  but that felt out of scope here (And isn't something we supported prior to this either).